### PR TITLE
Fix/various block editor

### PR DESCRIPTION
**Pull request by skogsmaskin**
* Upgrade Slate
* Better Word document cleaning logic
* Fix bug where an error occurred when clicking on a empty editor field (outside the content).
* Various smaller fixes.

### CODE REVIEW

File: packages/@sanity/form-builder/src/inputs/BlockEditor-slate/conversion/slateHTMLDeserializer/helpers.js [hunk 1]
```diff
@@ -21,20 +50,38 @@ function cleanUpWordDocument(html) {
     unwanted.parentNode.removeChild(unwanted)
   }
 
-  // Transform titles into H1s
-  const titleElments = document.evaluate(
-    "//p[@class='MsoTitle']",
+  // Transform mapped elements into what they should be mapped to
+  const mappedElements = document.evaluate(
+    `${mappedPaths.join('|')}`,
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/BlockEditor-slate/conversion/slateHTMLDeserializer/helpers.js]: No need to wrap this in a template string?

File: packages/@sanity/form-builder/src/inputs/BlockEditor-slate/conversion/slateHTMLDeserializer/helpers.js [hunk 2]
```diff
@@ -21,20 +50,38 @@ function cleanUpWordDocument(html) {
...
-    const title = titleElments.snapshotItem(i)
-    const h1 = document.createElement('h1')
-    h1.appendChild(new Text(title.textContent))
-    title.parentNode.replaceChild(h1, title)
+  for (let i = mappedElements.snapshotLength - 1; i >= 0; i--) {
+    const mappedElm = mappedElements.snapshotItem(i)
+    const mapToTag = elementMap[mappedElm.className]
+    const text = new Text(mappedElm.textContent)
+    let newElm
+    if (mapToTag.indexOf(':') > -1) {
```
Comment (bjoerge) [inline packages/@sanity/form-builder/src/inputs/BlockEditor-slate/conversion/slateHTMLDeserializer/helpers.js]: Use `includes(':')` instead

File: packages/@sanity/form-builder/src/inputs/BlockEditor-slate/conversion/slateHTMLDeserializer/helpers.js (hunk 1 above)
Comment (skogsmaskin) [reply packages/@sanity/form-builder/src/inputs/BlockEditor-slate/conversion/slateHTMLDeserializer/helpers.js]: Lol, no. There were once a need, but not anymore.

**Comment by skogsmaskin** [comment]
Thank you @bjoerge, I've adressed your nit picks!

**Comment by skogsmaskin** [comment]
Changed the base to 'master'. No need for going through 'next' here.

**Comment by skogsmaskin** [comment]
Rebased against 'master'

**Pull request merged**